pkg/scheduler/framework/plugins/dynamicresources/extended/util.go

```diff
@@ -22,17 +22,55 @@ import (
 	fwk "k8s.io/kube-scheduler/framework"
 )
 
+// DeviceClassMapping creates the mapping of extended resource name to device class name.
+// It always includes the implicit extended resource name for each device class.
 func DeviceClassMapping(draManager fwk.SharedDRAManager) (map[v1.ResourceName]string, error) {
+	deviceClassMap := make(map[v1.ResourceName]string)
+	mapping, err := extendedResourceToDeviceClassMapping(draManager)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range mapping {
+		deviceClassMap[k] = v.Name
+	}
+
+	return deviceClassMap, nil
+}
```

Comment:
I don't think we need 2 types of mappings for device class. One (in my opinion `map[v1.ResourceName]*resourceapi.DeviceClass`) should be enough.

Reply:
we can reduce it to one type of mappings for device class when there is the use case in the future, what do you think?

Reply:
I think that *resourceapi.DeviceClass is enough to access any its field, including name, so we don't need `resource` -> `class name` mapping.